### PR TITLE
Various small changes

### PR DESCRIPTION
**Pull request by leonschoorl**
- lots of small doc fixes
 - only export the constructors of `Reset` and `Enable` from `Clash.Signal.Internal`
 - warn about unparsed leftovers in json files
 - print location information of exceptions inside the compiler

### CODE REVIEW

**Comment by leonschoorl** [comment]
TODO: revert https://github.com/clash-lang/clash-compiler/pull/879

File: clash-lib/src/Data/Aeson/Extra.hs [hunk 1]
```diff
@@ -55,8 +58,13 @@ decodeOrErr
...
 decodeOrErr path contents =
   case parse json contents of
-    Done _ v ->
+    Done leftover v ->
       case fromJSON v of
+        Success a | BS.any notWhitespace leftover ->
+          trace ("WARNING: After a successfully parsing: " ++ show path
+                 ++ " there was some trailing garbage leftover, ignoring:\n"
+                 ++ BSChar.unpack leftover)
+                a
```
Comment (martijnbastiaan) [inline clash-lib/src/Data/Aeson/Extra.hs]: Although an improvement over the current situation, this should be a hard error IMO.

File: clash-prelude/src/Clash/Explicit/Signal.hs [hunk 2]
```diff
@@ -189,7 +189,7 @@ module Clash.Explicit.Signal
   , initBehavior
   , resetPolarity
     -- ** Enabling
-  , Enable(..)
+  , Enable
```
Comment (martijnbastiaan) [inline clash-prelude/src/Clash/Explicit/Signal.hs]: This should go into `CHANGELOG.md`

File: clash-prelude/src/Clash/Explicit/Signal.hs [hunk 3]
```diff
@@ -202,7 +202,7 @@ module Clash.Explicit.Signal
   , unsafeSynchronizer
   , veryUnsafeSynchronizer
     -- * Reset
-  , Reset(..)
+  , Reset
```
Comment (martijnbastiaan) [inline clash-prelude/src/Clash/Explicit/Signal.hs]: Likewise ^

File: clash-prelude/src/Clash/Signal.hs [hunk 4]
```diff
@@ -140,7 +140,7 @@ module Clash.Signal
   , unsafeSynchronizer
 #endif
     -- * Reset
-  , Reset(..)
+  , Reset
```
Comment (martijnbastiaan) [inline clash-prelude/src/Clash/Signal.hs]: Likewise ^

File: clash-prelude/src/Clash/Signal.hs [hunk 5]
```diff
@@ -153,7 +153,7 @@ module Clash.Signal
   , E.resetSynchronizer
   , holdReset
     -- ** Enabling
-  , Enable(..)
+  , Enable
```
Comment (martijnbastiaan) [inline clash-prelude/src/Clash/Signal.hs]: Likewise ^

**Pull request merged**